modules/tappxBidAdapter.js

```diff
@@ -261,11 +261,21 @@ function buildOneRequest(validBidRequests, bidderRequest) {
   imp.secure = 1;
 
   imp.bidfloor = utils.deepAccess(validBidRequests, 'params.bidfloor');
+  if (utils.isFn(validBidRequests.getFloor)) {
+    let sizes = (bannerMediaType !== undefined) ? bannerMediaType.sizes : videoMediaType.sizes;
+    sizes = (sizes !== undefined) ? sizes : '*';
+    imp.bidfloor = validBidRequests.getFloor({
+      currency: CUR,
+      bannerMediaType,
+      size: sizes
```

Comment:
The Price Floors module can be implemented on a per size floor, thus, this size param getFloor expects a single size array like so:

`[300, 250]`

You are passing it all the sizes which won't break it, but will just lead to it always picking the '*' floor.

If you want to get all the floors on per size, you can run getFloor for each size in your sizes array.

Then decide how you want to pass to your endpoint (maybe highest floor or smallest?)

Otherwise, if you do not wish to do this, you can just pass the '*' as the "size" param (or leave it out, same thing)



As for the `bennerMediaType` param you are passing.

This is expected to be a string representation of the mediaType.

You are passing the actual mediaTypeObject.

`banner` or `video` or `native` should be passed here.

Otherwise `*`

Here is a screenshot using your example:

![image](https://user-images.githubusercontent.com/13972794/116092355-96129600-a65a-11eb-8a02-3b005c4b1c19.png)


Also, it is reccomended to be a little safe when just assuming that the response from getFloor returns the object with the `.floor` attribute.

If not floor is found, your code here may throw an exception.

See how other adapters have surrounded this in a try / catch or even just validated the response has format expected.

Reply:
Hi! We update with the try/catch and change all with '*'.
We were to made some changes, but we just have problems to make a fake request with a priceFloor module to make some tests for this part.

Reply:
About the problem with testing...
We try to import the functions of priceFloors:
`import { handleSetFloorsConfig, getFloor } from 'modules/priceFloors';`
And create a configuration:
`const basicFloorConfig = {
      enabled: true,
      data: {
        // default if endpoint doesn't return in time
        floorProvider: 'floorProviderName',
        currency: 'USD',
        skipRate: 5,
        modelVersion: 'new model 1.0',
        schema: {
          fields: [ 'mediaType' ]
        },
        values: {
          'banner': 0.01,
          'video': 2.00
        }
      }
    }
    handleSetFloorsConfig({...basicFloorConfig, floorProvider: 'floorprovider'})`
But we don't know how to pass this information to a test fake bidrequest.
Any idea or trick to test this part?
Thanks for all.

Reply:
Hi @prebidtappx 

Please see how the rubicon bidAdapter mocked the getFloor response:

https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconBidAdapter_spec.js#L444

Note that we simply mock the responses and put a mock getFloor function onto the bidRequests.

See higher up how this mock is done where we use a global `getFloorResponse` object which we use to control what the mock function returns

https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconBidAdapter_spec.js#L300

See this line here how we add it on the bidRequest:

https://github.com/prebid/Prebid.js/blob/master/test/spec/modules/rubiconBidAdapter_spec.js#L446

Then throughout the test, we just update the global getFloorResponse var to be what we want to test and make sure our adapter handles correctly.

Reply:
Thanks for all!
We make a push with the testing.